java/src/main/native/src/TableJni.cpp

```diff
@@ -2655,25 +2655,28 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_partition(JNIEnv *env, jc
 
 JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_hashPartition(
     JNIEnv *env, jclass, jlong input_table, jintArray columns_to_hash, jint hash_function,
-    jint number_of_partitions, jintArray output_offsets) {
+    jint number_of_partitions, jint seed, jintArray output_offsets) {
 
   JNI_NULL_CHECK(env, input_table, "input table is null", NULL);
   JNI_NULL_CHECK(env, columns_to_hash, "columns_to_hash is null", NULL);
   JNI_NULL_CHECK(env, output_offsets, "output_offsets is null", NULL);
   JNI_ARG_CHECK(env, number_of_partitions > 0, "number_of_partitions is zero", NULL);
+  // For simplicity of converting a signed int to a unsigned int
+  JNI_ARG_CHECK(env, seed >= 0, "seed is negative", NULL);
```

Comment:
Not sure why this check is here.  Java doesn't have unsigned ints, so this eliminates half of the potential seeds we could use.  It's also not documented that passing a negative seed is an error.

Personally, I would just static cast the signed int into an unsigned int and pass it on, since then we get access to the full range of available seeds in the underlying algorithm.

Reply:
Nice suggestion, updated